app.go

```diff
@@ -275,6 +277,32 @@ type App struct {
 	logger       *fxlog.Logger
 	startTimeout time.Duration
 	stopTimeout  time.Duration
+	errorHooks   errorHookOption
+}
+
+// ErrorHook registers error handlers that implement error handling functions.
+// They are executed on invoke failures. Passing multiple ErrorHandlers appends
+// the new handlers to the application's existing list.
+func ErrorHook(funcs ...ErrorHandler) Option {
+	return errorHookOption(funcs)
+}
+
+// ErrorHandler implements Handle. They are used as error hooks and are called
+// on invoke errors.
+type ErrorHandler interface {
+	Handle(error)
```

Comment:
So I know I suggested that you rename this from `HandleError` to `Handle`, but an offline discussion with @abhinav made me realize that you had it right the first time :p

Naming this `Handle(error)` means that if you ever intend to have additional `Handle` methods, you'll lose naming consistency since the 'default' `Handle()` method is only for errors. Since structs implicitly implement interfaces and we don't have function overloading, calling this `Handle(error)` is likely undesirable.

sorry :/